### PR TITLE
 [Enhancement] add VSCode devcontainer configuration #116

### PR DESCRIPTION
**Pull request by dannyvassallo**
*Issue #, if available:*  116 

*Description of changes:*

* Adds a dev container with oh my zsh + aliases the team uses (with the bash equivalent)
* Adds instructions on how to use
* Adds rust-toolchain.toml to use rustup for rust and cargo version management

---
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# To get started with Dependabot version updates, you'll need to specify which
```
Comment (dannyvassallo) [inline .github/dependabot.yml:1]: this was autogenerated by vscode when i initialized the devcontainer

File: .devcontainer/.zshrc [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# If you come from bash you might have to change your $PATH.
+# export PATH=$HOME/bin:/usr/local/bin:$PATH
+
+# Path to your oh-my-zsh installation.
+export ZSH="$HOME/.oh-my-zsh"
```
Comment (joshfried-aws) [inline .devcontainer/.zshrc:5]: I'm not sure how i feel about something like this. How does this impact someone who doesnt have oh-my-zsh installed, or any of the themes/plugins here.  

Same with the aliases, what if we're overriding some aliases people already have and would rather not change?

File: README.md [hunk 3]
```diff
@@ -598,6 +599,11 @@ docker run \
 ```
 We should see the evaluation result emitted out on the console.
 
+### Tagging convention
+
+* We use the tag `latest` for the most recent docker image that gets published in sync with `main` branch of the `cloudformation-guard` GitHub repository.
```
Comment (joshfried-aws) [inline README.md:604]: We should probably mention tagged stable releases?

**Review by joshfried-aws — CHANGES_REQUESTED** [review]

File: .devcontainer/.zshrc (hunk 2 above)
Comment (dannyvassallo) [reply .devcontainer/.zshrc:5]: There's no conflict here because it's a container specifically for dev

**Review by joshfried-aws — APPROVED** [review]
Awesome, this will definitely make it easier for VSCode users who are interested in contributing!

**Review by razcloud — APPROVED** [review]

**Pull request merged**